pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,404 @@
+		{Name: "HEALTH_TIMEOUT_DATASTORE", Value: c.getHealthTimeoutDs()},
+		{Name: "HEALTH_PORT", Value: fmt.Sprintf("%d", DefaultHealthPort)},
+		{Name: "ICMP_PROBE_IPS", Value: icmpProbeIPs},
+		{Name: "ICMP_PROBE_INTERVAL", Value: icmpInterval},
+		{Name: "ICMP_PROBE_TIMEOUT", Value: icmpTimeout},
+		{Name: "HTTP_PROBE_URLS", Value: httpProbeURLs},
+		{Name: "HTTP_PROBE_INTERVAL", Value: httpInterval},
+		{Name: "HTTP_PROBE_TIMEOUT", Value: httpTimeout},
+		{Name: "EGRESS_POD_IP", ValueFrom: egressPodIp},
+		{Name: "EGRESS_VXLAN_VNI", Value: fmt.Sprintf("%d", c.config.EgressGWVxlanVNI)},
+		{Name: "LOG_SEVERITY", Value: c.config.EgressGW.GetLogSeverity()},
+	}
+}
+
+func (c *component) egwInitEnvVars() []corev1.EnvVar {
+	egressPodIp := &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"}}
+	return []corev1.EnvVar{
+		{Name: "EGRESS_VXLAN_VNI", Value: fmt.Sprintf("%d", c.config.EgressGWVxlanVNI)},
+		{Name: "EGRESS_VXLAN_PORT", Value: fmt.Sprintf("%d", c.config.EgressGWVxlanPort)},
+		{Name: "EGRESS_POD_IP", ValueFrom: egressPodIp},
+	}
+}
+
+func (c *component) egwPodSecurityPolicy() *policyv1beta1.PodSecurityPolicy {
+	boolTrue := true
+	psp := podsecuritypolicy.NewBasePolicy()
+	psp.GetObjectMeta().SetName("tigera-egress-gateway")
+	psp.Spec.AllowedCapabilities = []corev1.Capability{
+		corev1.Capability("NET_ADMIN"),
+	}
+	psp.Spec.AllowPrivilegeEscalation = &boolTrue
+	psp.Spec.HostIPC = true
+	psp.Spec.HostNetwork = true
+	psp.Spec.HostPID = true
+	psp.Spec.Privileged = true
+	psp.Spec.RunAsUser = policyv1beta1.RunAsUserStrategyOptions{
+		Rule: policyv1beta1.RunAsUserStrategyRunAsAny,
+	}
+	psp.Spec.SELinux = policyv1beta1.SELinuxStrategyOptions{
+		Rule: policyv1beta1.SELinuxStrategyRunAsAny,
+	}
+	psp.Spec.SupplementalGroups = policyv1beta1.SupplementalGroupsStrategyOptions{
+		Rule: policyv1beta1.SupplementalGroupsStrategyRunAsAny,
+	}
+	psp.Spec.FSGroup = policyv1beta1.FSGroupStrategyOptions{
+		Rule: policyv1beta1.FSGroupStrategyRunAsAny,
+	}
+	psp.Spec.Volumes = []policyv1beta1.FSType{
+		policyv1beta1.CSI,
+		policyv1beta1.Projected,
+	}
+	return psp
+}
+
+func (c *component) egwClusterRole() *rbacv1.ClusterRole {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups:     []string{"policy"},
+			Resources:     []string{"podsecuritypolicies"},
+			ResourceNames: []string{"tigera-egress-gateway"},
+			Verbs:         []string{"use"},
+		},
+	}
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "tigera-egress-gateway",
+		},
+		Rules: rules,
+	}
+}
+
+func (c *component) egwClusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.config.EgressGW.Name,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "tigera-egress-gateway",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "tigera-egress-gateway",
```

Comment:
Can there be more than 1 egw in the same namespace and is the name always this value?
Otherwise, this should be EgressGW.name